test/native/integration-test-helpers/with-fake-timers.js

```diff
@@ -18,6 +18,18 @@ export async function withFakeTimers( fn ) {
 		jest.useFakeTimers( { legacyFakeTimers: true } );
 	}
 
+	// `Date.now` returns the real-time even when using fake timers.
+	// Some functions like `debounce` relies on the time returned by this function
+	// for its calculations, which can lead to wrong behaviors when executing timers.
+	// To avoid this, we mock this function and return the time provided Jest fake timers.
+	// Reference: https://jestjs.io/docs/jest-object#jestnow
+	let dateNowSpy;
+	if ( ! jest.isMockFunction( Date.now ) ) {
+		dateNowSpy = jest
+			.spyOn( Date, 'now' )
+			.mockImplementation( () => jest.now() );
+	}
```

Comment:
Are we already or have we explored using Jest's "modern" fake timers instead of manually mocking `Date.now`? 

It appears modern timers are the official answer to this `Date.now` problem: [intent](https://github.com/jestjs/jest/issues/2684#issuecomment-421098542) to fix, [adjustment](https://github.com/jestjs/jest/pull/5171#issuecomment-528752754) to the proposed solution, and [completion](https://github.com/jestjs/jest/pull/7776) of the solution. 

Using modern timers [caused issues](https://github.com/WordPress/gutenberg/pull/37715#discussion_r781214591) in our tests in the past. I believe that issue may be [resolved](https://github.com/facebook/react-native/commit/f1fdc8b9b636e010ae5d5e737fc81a8da8f536d4) in React Native v0.71.0. If we are still unable to leverage modern timers now, maybe we should put an inline comment in this file that we could explore removing the manual `Date.now` mock after we upgrade to at least React Native v0.71.0?

Reply:
> Are we already or have we explored using Jest's "modern" fake timers instead of manually mocking `Date.now`?

I tried using "modern" fake timers but tests exceed the maximum time out. We'd need to investigate further what are the blockers to use this.

> It appears modern timers are the official answer to this `Date.now` problem: [intent](https://github.com/jestjs/jest/issues/2684#issuecomment-421098542) to fix, [adjustment](https://github.com/jestjs/jest/pull/5171#issuecomment-528752754) to the proposed solution, and [completion](https://github.com/jestjs/jest/pull/7776) of the solution.

👍 

> Using modern timers [caused issues](https://github.com/WordPress/gutenberg/pull/37715#discussion_r781214591) in our tests in the past. I believe that issue may be [resolved](https://github.com/facebook/react-native/commit/f1fdc8b9b636e010ae5d5e737fc81a8da8f536d4) in React Native v0.71.0. If we are still unable to leverage modern timers now, maybe we should put an inline comment in this file that we could explore removing the manual `Date.now` mock after we upgrade to at least React Native v0.71.0?

Sure, I can add a note mentioning that this workaround could be removed once we upgrade to `0.71.0`.

Reply:
Thank you for exploring. Moving forward with this workaround makes sense to me. 👍🏻